File: src/test/java/integration/auth/validators/PasswordValidatorTest.java
package integration.auth.validators;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.matag.admin.auth.validators.PasswordValidator;
import com.matag.admin.auth.validators.ValidationException;

public class PasswordValidatorTest {
  private final PasswordValidator passwordValidator = new PasswordValidator();

  @Test
  public void validPassword() {
    passwordValidator.validate("valid");
  }

  @Test
  public void tooShortPassword() {
    Assertions.assertThrows(ValidationException.class, () ->
      passwordValidator.validate("1")
    );
  }
}

File: src/test/java/integration/auth/validators/PasswordValidatorTest.kt
package integration.auth.validators

import com.matag.admin.auth.validators.PasswordValidator
import com.matag.admin.auth.validators.ValidationException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable

class PasswordValidatorTest {
    private val passwordValidator = PasswordValidator()

    @Test
    fun validPassword() {
        passwordValidator.validate("valid")
    }

    @Test
    fun tooShortPassword() {
        Assertions.assertThrows(
            ValidationException::class.java
        ) { passwordValidator.validate("1") }
    }
}